src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php

```diff
@@ -276,18 +262,41 @@ public function getRouteParams()
     }
 
     /**
-     * Gets the controller.
+     * Gets the parsed controller.
      *
-     * @return string The controller as a string
+     * @return array|string The controller as a string or array of data
+     *                      with keys 'class', 'method', 'file' and 'line'
      */
     public function getController()
     {
         return $this->data['controller'];
     }
 
+    /**
+     * Gets the parsed forward controller.
+     *
+     * @return array|bool An array with keys 'token' the forward profile token, and
+     *                    'controller' the parsed forward controller, false otherwise
+     */
+    public function getForward()
+    {
+        return isset($this->data['forward']) ? $this->data['forward'] : false;
+    }
+
+    public function getRedirect()
+    {
+        return isset($this->data['redirect']) ? $this->data['redirect'] : false;
+    }
+
     public function onKernelController(FilterControllerEvent $event)
     {
         $this->controllers[$event->getRequest()] = $event->getController();
+
+        if ($parentRequestAttributes = $event->getRequest()->attributes->get('_forwarded')) {
+            if ($parentRequestAttributes instanceof ParameterBag) {
+                $parentRequestAttributes->set('_forward_controller', $this->parseController($event->getController()));
```

Comment:
I don't think it's need to test if `parentRequest->has('_forward_controller')` before overriding it.

Forward action should be use to delegate the return of the response to another controller as it also duplicate the original request. What would be the use case for calling it after the $response has been set ? There is no point to duplicate original request twice IMO.

Though overriding `_forward_controller` value seems a logical behaviour, we could even through an exception from abstract controller in framework bundle like my example of Silex usage in the description.